packages/smart/funcs.go

```diff
@@ -2383,3 +2385,50 @@ func PubToHex(in interface{}) (ret string) {
 	}
 	return
 }
+
+// ExternalNetInfo describes external blockchains
+type ExternalNetInfo struct {
+	URL       string `json:"url"`       // Node address
+	Contract  string `json:"contract"`  // The name of the contract
+	Condition string `json:"condition"` // Access rights
+	Interval  string `json:"interval"`  // Interval of sending data
+}
+
+func SendToNetwork(sc *SmartContract, netName string, params interface{}) (err error) {
+	var (
+		out, insertQuery string
+		external         map[string]ExternalNetInfo
+		netInfo          ExternalNetInfo
+		ok               bool
+	)
+	out, err = JSONEncode(params)
+	if err != nil {
+		return err
+	}
+	if err = unmarshalJSON([]byte(syspar.SysString(syspar.ExternalBlockchain)), &external,
```

Comment:
Let's move this to the cache. We will only unmarshall when starting and updating the parameter.

Reply:
SysString gets the value from the cache. 
This function is not called too often. So, I don't think that unmarshalJSON slows down.